File: ReactNativeApp/android/app/src/main/java/com/components/detection/ball/BallBuilder.java
package com.components.detection.ball;

import com.models.ModelManager;
import com.ssvc.SSVC;
import com.ssvc.SSVCBuilder;

public class BallBuilder extends SSVCBuilder {
    static BallDetectionState buildState(ModelManager.MODELTYPE modelType, boolean exerciseLead) {
        return new BallDetectionState(modelType, exerciseLead);
    }

    public static SSVC<BallDetectionState, BallDetectionView, BallDetectionController> buildBallDetection(ModelManager.MODELTYPE modelType, boolean exerciseLead) {
        BallDetectionState state = buildState(modelType, exerciseLead);
        BallDetectionView ballStateView = new BallDetectionView(state);
        BallDetectionController ballController = new BallDetectionController(state);
        return new SSVC<>(state, ballStateView, ballController);
    }
}

File: ReactNativeApp/ios/Modules/SSVC/Instance/Detection/Ball/BallBuilder.swift
//
//  BallBuilder.swift
//  jogo
//
//  Created by arham on 04/11/2021.
//

import Foundation


class BallBuilder: SSVCBuilder {
  
  internal static func buildState(modelType: ModelManager.MODELTYPE, exerciseLead: Bool) -> BallState {
    return BallState(modelType: modelType, exerciseLead: exerciseLead, observerId: Football.ObserverID.FOOTBALL.rawValue)
  }
  
  internal static func build(state: BallState) -> BallSSVC<BallState, BallView, BallController> {
    let view = BallPointView(state: state)
    let controller = BallController(state: state)
    return BallSSVC(state: state, stateView: view, controller: controller)
  }
  
  internal static func build(modelType: ModelManager.MODELTYPE, exerciseLead: Bool) -> BallSSVC<BallState, BallView, BallController> {
    let state = buildState(modelType: modelType, exerciseLead: exerciseLead)
    let view = BallPointView(state: state)
    let controller = BallController(state: state)
    return BallSSVC(state: state, stateView: view, controller: controller)
  }
  
}
